### ISSUE TITLE
CMAKE_INSTALL_LIBDIR not respected for cmake files

### ISSUE THREAD

**Issue by Arfrever** [libssh2/libssh2#405]
When `CMAKE_INSTALL_LIBDIR` is set to `lib64`, most files are installed correctly, but 3 files are incorrectly installed in hardcoded `lib` directory.

Fix:
```
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -397,7 +397,7 @@ set(RUNTIME_DEPENDENCIES ${_RUNTIME_DEPENDENCIES} CACHE INTERNAL
 ## During package installation, install Libssh2Config.cmake
 install(EXPORT Libssh2Config
   NAMESPACE Libssh2::
-  DESTINATION lib/cmake/libssh2)
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/libssh2)
 
 ## During build, register directly from build tree
 # create Libssh2Config.cmake
@@ -429,4 +429,4 @@ write_basic_package_version_file(
   COMPATIBILITY SameMajorVersion)
 install(
   FILES ${CMAKE_CURRENT_BINARY_DIR}/Libssh2ConfigVersion.cmake
-  DESTINATION lib/cmake/libssh2)
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/libssh2)
```

**Comment by willco007** [comment]
Seems safe enough, anyone have comments on this one?

**Comment by tiennou** [comment]
LGTM, for what it's worth, I'm okay with this going in.

**Comment by LooTKinGz** [comment]
Repository update

**Comment by Arfrever** [comment]
Ping.

**Comment by willco007** [comment]
fixed with #405.
